bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkMutation.java

```diff
@@ -416,28 +417,35 @@ public synchronized void run() {
       } finally {
         addCallback(mutateRowsFuture, operationId);
       }
+      setupStalenessChecker();
     }
 
-    /**
-     * A {@link ComplexOperationStalenessHandler} is a method of last resort in cases where somehow the various
-     * listeners failed to get a response, or had some kind of strange issue.
-     */
-    private ComplexOperationStalenessHandler createRetryHandler() {
-      return new ComplexOperationStalenessHandler() {
+    private void setupStalenessChecker() {
+      if (currentRequestManager == null){
```

Comment:
We also check for empty() below. Is that necessary here?  If so does it help to have this check be a method with a self-documenting name?